### PR TITLE
Share more code in the 'use collection expression' fixers

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/69449.

### CODE REVIEW

File: src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForArrayCodeFixProvider.cs [hunk 1]
```diff
@@ -25,130 +25,99 @@ namespace Microsoft.CodeAnalysis.CSharp.UseCollectionExpression;
 using static UseCollectionExpressionHelpers;
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForArrayCodeFixProvider.cs:25]: view with whitespace off.

File: src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForArrayCodeFixProvider.cs [hunk 2]
```diff
@@ -25,130 +25,99 @@ namespace Microsoft.CodeAnalysis.CSharp.UseCollectionExpression;
...
-        // We're going to be continually editing this tree.  Track all the nodes we
-        // care about so we can find them across each edit.
-        var semanticDocument = await SemanticDocument.CreateAsync(
-            document.WithSyntaxRoot(originalRoot.TrackNodes(arrayExpressions)),
-            cancellationToken).ConfigureAwait(false);
-
-        while (arrayExpressions.Count > 0)
+        if (arrayCreationExpression is InitializerExpressionSyntax initializer)
         {
-            var arrayCreationExpression = semanticDocument.Root.GetCurrentNodes(arrayExpressions.Pop()).Single();
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForArrayCodeFixProvider.cs:74]: all this logic moved into the base class.

File: src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForStackAllocCodeFixProvider.cs [hunk 3]
```diff
@@ -21,111 +19,76 @@
...
+            {
+                StackAllocArrayCreationExpressionSyntax arrayCreation => arrayCreation.WithInitializer(i),
+                ImplicitStackAllocArrayCreationExpressionSyntax implicitArrayCreation => implicitArrayCreation.WithInitializer(i),
+                _ => throw ExceptionUtilities.Unreachable(),
+            },
             cancellationToken).ConfigureAwait(false);
 
-        while (stackallocExpressions.Count > 0)
-        {
-            var stackAllocExpression = semanticDocument.Root.GetCurrentNodes(stackallocExpressions.Pop()).Single();
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForStackAllocCodeFixProvider.cs:70]: all moved into base class.

File: src/Analyzers/Core/CodeFixes/UseCollectionInitializer/AbstractUseCollectionInitializerCodeFixProvider.cs [hunk 4]
```diff
@@ -62,80 +65,42 @@ public sealed override ImmutableArray<string> FixableDiagnosticIds
...
-            }
-
             var solutionServices = document.Project.Solution.Services;
 
-            // We're going to be continually editing this tree.  Track all the nodes we
-            // care about so we can find them across each edit.
-            var semanticDocument = await SemanticDocument.CreateAsync(
-                document.WithSyntaxRoot(originalRoot.TrackNodes(originalObjectCreationNodes.Select(static t => t.objectCreationExpression))),
-                cancellationToken).ConfigureAwait(false);
-
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/UseCollectionInitializer/AbstractUseCollectionInitializerCodeFixProvider.cs:107]: all moved into base class.

**Review by akhera99 — APPROVED** [review]

**Pull request merged**